[feature/StoryLog] 로딩 효과 수정

Kkumteulyi opened the pull request:
## 개요
<!-- 어떤 점이 변경되었는지, 간단하게 작성해주세요. -->
### 1. 로딩 씬 수정
- 부자연스럽게 연출되던 로딩을 수정하였습니다
### 2. 로딩 관련 코드 리팩토링
- 기존에 2 중 for 문을 사용하거나 불필요한 오브젝트를 추가로 호출하던 코드를 수정하였습니다 
## 변경 사항
<!-- 어떤 점에 변경되었는지 상세하게 작성해주세요.-->
## LoadingEffectManager.cs
### 1. Awake(코드 호출 시)
- **GetComponetsInChildren**활용하여 for문을 사용하지 않도록 변경하였습니다
- 위, 아래 레이어를 불러와 각각의 Component를 할당하던 방식 대신 로딩 씬 오브젝트만 불러오도록 수정하였습니다
- 원하는 레이어 순서를 만들기 위해 **LINQ**를 활용하였습니다
- **layerCount**도 현재 층마다 동일한 레이어가 존재하므로 childCount를 사용하여 초기화 하도록 변경하였습니다

### FadeIn,FadeOut
- Sequence가 끝나면 다음 오브젝트의 Sequence가 진행되도록 수정했던 것을 롤백 했습니다
-> WaitforSecond(**delay = 전체 시간 / 레이어 오브젝트 수**)를 통해 자연스럽게 연출 되도록 복구
- SetEase(Ease.Linear)를 사용하여 일정하게 변하도록 설정
=> 설정을 하지 않으면 Ease.OutQuad가 기본 값

### TutorialManager.cs
- FadeOut 효과 추가(튜토리얼 종료 시, 스킵 버튼 누를 시)

변경된 오브젝트
- Loading Scene Canvas(태그 추가 및 기존 씬에서 가져옴)
=> 연출에 직접적으로 사용되는 레이어들 **Prefab**으로 변경
- Player Information Section
- Log Scroll View(사이즈가 안맞는 것 같아 가져옴)
## 참고 자료
<!-- 기능 개발에 참고한 자료가 있다면 작성해주세요. (필수는 아닙니다.) -->

Kkumteulyi committed:
로딩 씬 수정

부자연스러운 로딩 수정

Kkumteulyi committed:
LoadingEffectManager에 불필요한 코드 제거

FadeIn을 호출하며 시작하므로 alpha는 초기화 하지 않아도 될 것 같아 초기에 alpha를 지우는 코드를 제거하였습니다

miniron-v reviewed (changes requested):
페이드 인/아웃이 이전만큼 부드럽지 않고, 왼쪽에 나타나는 것도 이상하네요.

Kkumteulyi committed:
프리팹 수정, 연출 수정

프리팹 수정
- Rhombus 1을 사용하는 부모 레이어를 프리팹으로 만들고 불필요해진 Rhombus 1 프리팹은 제거
- 로딩 씬에 사용되던 레이어 프리팹으로 교체

연출 수정
- 오브젝트 하나의 연출이 끝나면 다음 오브젝트 연출을 시작하던 방식은 부자연스러우므로 이전 설정으로 롤백

Kkumteulyi committed:
튜토리얼 씬에 FadeOut 추가

튜토리얼 종료 후 화면 전환 연출 추가

Kkumteulyi committed:
튜토리얼 스킵 버튼 시 로딩 효과 추가

튜토리얼 스킵 버튼 누르면 로딩 되도록 효과를 추가했습니다

miniron-v reviewed (changes requested):
async를 사용한 건 좋았으나, 로딩 효과 구조가 **페이드 아웃 -> 이벤트 변경 및 처리 -> 페이드 인**에서 **페이드 아웃 -> 페이드 인 -> 이벤트 변경 및 처리**로 작업된 게 아쉽네요. 이벤트 변경 코스트가 낮아서 티는 안 날 수 있지만 결국 타이밍은 안 맞는 거고, 혹시나 코스트가 높아지면 금세 티나서 다시 고치게 될 겁니다.

이렇게 **이전 작업을 다시 고치는** 수요가 늘어나면 늘어날수록, 개발 속도도 더뎌지고 개발자는 훨씬 힘들어질 거예요. 그래서 다시 수정 요청 드리겠습니다.

Kkumteulyi committed:
Async 대신 코루틴으로 변경

Async와 코루틴을 혼합하여 사용하는 방식은 좋지 않다고 생각되어 코루틴을 사용하도록 변경하였습니다

Kkumteulyi commented:
## 변경 사항
### TutorialManager.cs
- **SkipEvent**에 코루틴 함수인 FadeOut, FadeIn을 추가하기 위해 기존 함수에 있던 내용을 따로 분리하여 
**SkipEventCo** 코루틴 함수를 생성하여 이동 시켰습니다
- DialogueManager.cs에서 현재 사용하는 **fadeSpeed** 변수의 값이 0.5f이므로 동일하게 변경하였습니다
### StoryInformation.cs
- 기존에 duration/2 와 같이 함수 내부에서 전달 받은 속도 인자를 임의로 조정하여 사용하던 방식 대신 duration으로 변경하였습니다
- 스토리 이름이 1초 동안 보이도록 WaitForSeconds를 사용하였습니다
### DialogueManager.cs
- 스토리 이름을 띄워주는 함수에 fadeSpeed*2와 같이 임의로 변수 값을 조정하는 대신 fadeSpeed를 그대로 전달하도록 변경하였습니다

miniron-v reviewed (approved):
며칠 간 수정하느라 고생 많으셨습니다. 몇 가지 당부 사항만 적고 머지할게요.

### 1. 다른 역할에 같은 변수를 쓰지 말자
- fadeSpeed는 FadeIn/FadeOut 시간으로 쓰이는 변수입니다. (사실 fadeTime이 더 잘 어울리긴 하죠.)
- ShowInformation은 시작된 이벤트의 이름을 적어주는 함수고, 그 시간을 정하기 위한 변수가 필요하죠.
- 이러면 **당연히 새 변수를 만들어야** 합니다. 단순히 **둘의 값이 같으니 같이 쓰자**가 되면, 둘 중 하나의 시간이 달라지는 순간 나머지도 다 엇갈리게 됩니다.
- 지금이 1KB 안에 게임을 만드는 80년대면 모를까, 티 안 나는 성능 향상을 위해 협업과 확장성을 망가뜨리는 일은 없었으면 좋겠습니다.
